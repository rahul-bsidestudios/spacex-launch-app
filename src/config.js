// Place all the app level constants and configurations here

export const API_URL = "https://api.spacexdata.com/v3/launches?limit=100";

export const START_LAUNCH_YEAR = 2006;

export const END_LAUNCH_YEAR = 2020;