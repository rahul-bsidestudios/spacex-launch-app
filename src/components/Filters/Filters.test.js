import React from "react";
import { shallow } from "enzyme";
import { Filters } from "./index";
import { START_LAUNCH_YEAR } from "../../config";

describe("Test cases for Filters component", () => {
	const mockProps = {
		onFilterChange: jest.fn()
	};
	const component = shallow(<Filters { ...mockProps } />);
	
	test("Should match snapshot", () => {
		expect(component).toMatchSnapshot();
	});

	test("Should call onFilterChange if any filter is clicked", () => {
		component.find("li.filter").at(0).find("button").simulate("click");
		expect(mockProps.onFilterChange).toHaveBeenCalledWith({ launch_year: START_LAUNCH_YEAR.toString() });
	});
});