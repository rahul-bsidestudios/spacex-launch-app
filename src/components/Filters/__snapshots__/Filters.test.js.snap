// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases for Filters component Should match snapshot 1`] = `ShallowWrapper {}`;
